client/src/modules/cash/payments/templates/search.modal.html

```diff
@@ -55,6 +55,31 @@
 
           <div class="form-group">
             <label class="control-label" translate>
+              FORM.LABELS.DEBTOR_GROUP
+            </label>
+
+            <!-- @todo make this a component - it is used everywhere. Calls back method
+                 `clear` and resets the id passed to the component -->
+            <span style="display:inline-block;" class="pull-right">
+              <a href ng-click="$ctrl.clear('debtor_group_uuid')" tabindex="-1">
+                <i class="fa fa-eraser"></i> <span translate>FORM.BUTTONS.CLEAR</span>
+              </a>
+            </span>
+
+            <!-- @todo make data driven selects components to handle loading and error states -->
+            <select
```

Comment:
In most places, I think we use a `ui-select` for debtor groups.  We should probably use the same here.